src/randgraphs.jl

```diff
@@ -193,6 +193,191 @@ function barabasi_albert(n::Integer, k::Integer; is_directed=false, seed::Int=-1
+        # skip if loop edge
+        (source == target) && continue
+        edge = Edge(source, target)
+        # is there already an edge? If so, try again
+        in(edge, edges) && continue
+        # insert the edge
+        push!(edges, edge)
+        m -= 1
+    end
+    # create the graph
+    g = DiGraph(n)
+    for edge in edges
+        add_edge!(g, edge)
+    end
+    return g
+end
+
+"""
+    function static_scale_free(n::Int, m::Int, alpha::Float64; seed::Int=-1, finite_size_correction::Bool=false)
+
+Generates a non-growing random graph with expected power-law degree distributions. 
+`n` is the number of nodes in the generated graph. `m` is the number of edges in the generated graph. 
+`alpha` is the power law exponent of digree distribution. 
+`finite_size_correction` determines whether to use the proposed finite size correction of Cho et al. 
+Time complexity is Time complexity is O(|V| + |E| log |E|).
+
+References:
+
+* Goh K-I, Kahng B, Kim D: Universal behaviour of load distribution in scale-free networks. Phys Rev Lett 87(27):278701, 2001.
+
+* Chung F and Lu L: Connected components in a random graph with given degree sequences. Annals of Combinatorics 6, 125-145, 2002.
+
+* Cho YS, Kim JS, Park J, Kahng B, Kim D: Percolation transitions in scale-free networks under the Achlioptas process. Phys Rev Lett 103:135702, 2009.
+"""
+function static_scale_free(n::Int, m::Int, alpha::Float64; seed::Int=-1, finite_size_correction::Bool=false)
+    @assert(n >= 0, "Invalid number of nodes")
+    @assert(alpha >= 2., "out-degree exponent must be >= 2")
+    alpha = -1.0/(alpha-1.)
+    # construct the fitness
+    fitness = zeros(n)
+    j = float(n)
+    if finite_size_correction && alpha < -0.5
+        # See the Cho et al paper, first page first column + footnote 7
+        j += n^(1.+0.5/alpha) * (10.*sqrt(2.)*(1.+alpha))^(-1./alpha)-1.
+    end
+    if j < n
+        j = float(n)
+    end
+    for i in eachindex(fitness)
+        fitness[i] = j^alpha
+        j -= 1.
+    end
+    static_fitness_model(m, fitness, seed=seed)
+end
+
+function static_scale_free(n::Int, m::Int, alpha_out::Float64, alpha_in::Float64; seed::Int=-1, finite_size_correction::Bool=false)
+    @assert(n >= 0, "Invalid number of nodes")
+    @assert(alpha_out >= 2., "out-degree exponent must be >= 2")
+    @assert(alpha_in >= 2., "in-degree exponent must be >= 2")
+    alpha_out = -1.0/(alpha_out-1.)
+    alpha_in = -1.0/(alpha_in-1.)
+    # construct the fitness
+    fitness_out = zeros(n)
+    j = float(n)
+    if finite_size_correction && alpha_out < -0.5
+        # See the Cho et al paper, first page first column + footnote 7
+        j += n^(1.+0.5/alpha_out) * (10.*sqrt(2.)*(1.+alpha_out))^(-1./alpha_out)-1.
+    end
+    if j < n
+        j = float(n)
+    end
+    for i in eachindex(fitness_out)
+        fitness_out[i] = j^alpha_out
+        j -= 1.
+    end
+    fitness_in = zeros(n)
+    j = float(n)
```

Comment:
not true, without `float` `j` would be an Int after this line. For type stability better be cautious